packages/noya-state/src/reducers/canvasReducer.ts

```diff
@@ -593,6 +636,46 @@ export function canvasReducer(
 
             break;
           }
+          case 'maybeMoveGradientElipseLength': {
+            return;
+          }
+          case 'moveGradientElipseLength': {
+            const { current } = interactionState;
+            if (!state.selectedGradient) return;
+
+            const { layerId, fillIndex } = state.selectedGradient;
+
+            const indexPath = Layers.findIndexPath(
+              page,
+              (layer) => layer.do_objectID === layerId,
+            );
+
+            if (!indexPath) return;
+
+            const draftLayer = Layers.access(
+              draft.sketch.pages[pageIndex],
+              indexPath,
+            );
+
+            if (
+              draftLayer.style?.fills?.[fillIndex]?.fillType !==
+                Sketch.FillType.Gradient &&
+              draftLayer.style?.fills?.[fillIndex]?.gradient.gradientType !==
+                Sketch.GradientType.Radial
+            )
+              return;
+            const gradient = draftLayer.style?.fills?.[fillIndex]?.gradient;
+            const points = getSelectedGradientStopPoints(state);
+            if (!points) return;
+            const center = points[0].point;
+            const lastPoint = points[points.length - 1].point;
+
+            const radius = distance(center, lastPoint);
+            const length = distance(center, { x: current.x, y: center.y });
```

Comment:
At larger distances, the movement jumps right at the start of the drag, and is slightly different than the mouse position.

I think this should fix it:

```ts
const length = distance(current, getClosestPointOnLine(current, [center, lastPoint]))
```